lua/iswap/internal.lua

```diff
@@ -47,44 +47,54 @@ function M.get_list_node_at_cursor(winid, config)
     local start_row, start_col, end_row, end_col = node:range()
     local start = { start_row, start_col }
     local end_ = { end_row, end_col }
-    if util.within(start, cursor_range, end_) and node:named_child_count() > 0 then
-      ret = node
+    if util.within(start, cursor_range, end_) and node:named_child_count() > 1 then
+      local children = ts_utils.get_named_children(node)
+      if find_cur_node then
+        local cur_nodes = util.nodes_containing_cursor(children, winid)
+        if #cur_nodes >= 1 then
+          if #cur_nodes > 1 then
+            err("multiple found, using first", config.debug)
+          end
+          ret = { node, children, cur_nodes[1] }
+        end
+      else
+        ret = { node, children }
```

Comment:
Could you refactor this by first setting `ret` and then conditionally doing `table.insert(ret, cur_nodes[1]` in an `if`?

Reply:
Actually no, I just realised that ret shouldn't be unconditionally set before the `if find_cur_node`. We specifically want to exclude nodes where we aren't on one of the children. I don't think any refactor here would be clearer than the current way

Reply:
basically if find_cur_node is true and cur_nodes is empty, that means the cursor happens to lie in the parent node (the list) but not in any of the children. This happens if you are on the braces of a list for example. It doesn't make sense to use this list nodes for ISwapWith, because there is no cur_node, so we just skip it. However we can still use this list for ISwap because the user will use the labels to select the two nodes.

Reply:
Ahhh, I see. Can we change the parameter name to `needs_cursor_node`?